ofrak_core/ofrak/core/cpio.py

```diff
@@ -85,23 +86,39 @@ class CpioUnpacker(Unpacker[None]):
     external_dependencies = (CPIO_TOOL,)
 
     async def unpack(self, resource: Resource, config=None):
+
+        resource_data = await resource.get_data()
         cpio_v = await resource.view_as(CpioFilesystem)
-        resource_data = await cpio_v.resource.get_data()
+
         with tempfile.TemporaryDirectory() as temp_flush_dir:
+
+            temp_file_path = os.path.join(temp_flush_dir, "temp_cpio")
```

Comment:
Instead of putting the CPIO in the directory we're going to extract to, we should probably use `tempfile.TemporaryFile` to make a new one that is already opened with `'wb'`, and will delete automatically.